CSIP4: Create testCase.xml

koit opened the pull request.

koit committed:
Create testCase.xml for CSIP4

koit committed:
Improve testCase.xml for CSIP4

Add description for dependency.
Shorten corpus package names.
Straighten quotes.

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP4/testCase.xml:
```diff
@@ -1,23 +1,161 @@
-  <id specification="E-ARK CSIP" version="2.0-DRAFT" requirementId="CSIP4"/>
-  <!-- URL references to requirements for convenient lookup -->
+<testCase xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="testCase.xsd" testable="TRUE">
+  <id specification="CSIP" version="2.0.3" requirementId="CSIP4"/>
```
new version is 2.0.4

koit replied:
Version fixed.

PhillipTommerholt replied:
Is says 2.0.3 and not 2.0.4?

koit replied:
Must be some caching issue with GitHub GUI - I fixed the version in commit c58aa8a2dad562257bf289d9e9e6c8b3ca345081. When I click the "Files changed" tab, it shows 2.0.4.

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP4/testCase.xml:
```diff
@@ -1,23 +1,161 @@
+      </description>
+      <error level="WARNING">
+        <message>
+          mets/@csip:CONTENTINFORMATIONTYPE attribute does not exist. SHOULD be: mets/@csip:CONTENTINFORMATIONTYPE exists and has a value from VocabularyContentInformationTypeSpecification.
```
Maybe this message:
mets/@csip:CONTENTINFORMATIONTYPE attribute does not exist. A mets/@csip:CONTENTINFORMATIONTYPE SHOULD exist and SHOULD have a value from VocabularyContentInformationTypeSpecification.

koit replied:
I agree that the message style you propose might be easier to read. I pondered the alternatives quite a bit in the beginning and then chose this style (i.e. "Problem description. Correct situation description."). I suggest we leave it as it is and during the final review of all testCases we look at the different styles, figure out the best one and then rewrite all testCase.xml to have uniform style. In my view, the developers should use the error messages from our testCase.xml in their software, so having a unified style is important, but I don't have a clear style preference yet.

PhillipTommerholt replied:
I agree!
Speaking from our experience there is an extreme value in doing this right and putting some effort in it. 
Users of the validation tool need to understand the problems that the validation tool finds.

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP4/testCase.xml:
```diff
@@ -1,23 +1,161 @@
+      </description>
+      <error level="ERROR">
+        <message>
+          mets/@csip:CONTENTINFORMATIONTYPE attribute does not exist. MUST be: mets/@csip:CONTENTINFORMATIONTYPE exists and has a value from VocabularyContentInformationTypeSpecification.
```
Maybe this message:
mets/@csip:CONTENTINFORMATIONTYPE attribute does not exist. A mets/@csip:CONTENTINFORMATIONTYPE MUST exist and MUST have a value from VocabularyContentInformationTypeSpecification.

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP4/testCase.xml:
```diff
@@ -1,23 +1,161 @@
+      <description>If mets/@csip:CONTENTINFORMATIONTYPE exists, it MUST have a value from the VocabularyContentInformationTypeSpecification as defined by DILCIS Board.</description>
+      <error level="ERROR">
+        <message>
+          mets/@csip:CONTENTINFORMATIONTYPE attribute has incorrect value. MUST be: mets/@csip:CONTENTINFORMATIONTYPE has a value from the VocabularyContentInformationTypeSpecification as defined by DILCIS.
```
What about the message:
mets/@csip:CONTENTINFORMATIONTYPE attribute has an incorrect value. 
mets/@csip:CONTENTINFORMATIONTYPE MUST have a value from the VocabularyContentInformationTypeSpecification.

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP4/testCase.xml:
```diff
@@ -1,23 +1,161 @@
+      <description>If mets/@csip:CONTENTINFORMATIONTYPE = "OTHER" then mets/@csip:OTHERCONTENTINFORMATIONTYPE MUST exist.</description>
+      <error level="ERROR">
+        <message>
+          mets/@csip:OTHERCONTENTINFORMATIONTYPE attribute does not exist. MUST be: If mets/@csip:CONTENTINFORMATIONTYPE = "OTHER" then mets/@csip:OTHERCONTENTINFORMATIONTYPE MUST exist and have a value.
```
What about this message:

mets/@csip:OTHERCONTENTINFORMATIONTYPE attribute does not exist. If mets/@csip:CONTENTINFORMATIONTYPE = "OTHER" then mets/@csip:OTHERCONTENTINFORMATIONTYPE MUST exist and have a value.

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP4/testCase.xml:
```diff
@@ -1,23 +1,161 @@
+      <description>If mets/@csip:CONTENTINFORMATIONTYPE = "OTHER" then mets/@csip:OTHERCONTENTINFORMATIONTYPE MUST exist and have a value.</description>
+      <error level="ERROR">
+        <message>
+          mets/@csip:OTHERCONTENTINFORMATIONTYPE attribute has no value. MUST be: If mets/@csip:CONTENTINFORMATIONTYPE = "OTHER" then mets/@csip:OTHERCONTENTINFORMATIONTYPE MUST exist and have a value.
```
What about this message:
mets/@csip:OTHERCONTENTINFORMATIONTYPE attribute has no value. If mets/@csip:CONTENTINFORMATIONTYPE = "OTHER" then mets/@csip:OTHERCONTENTINFORMATIONTYPE MUST exist and have a value.

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP4/testCase.xml:
```diff
@@ -1,23 +1,161 @@
+          <path></path>
+          <description>
+            METS.xml file is located in the /representations folder.
+            The METS.xml file attribute mets/@csip:CONTENTINFORMATIONTYPE does not exist.
```
I think that the package needs to have missing attributes at both root and representation level to make sure that we see the expected different behavior on both levels... (warning at root level and error at representation level).

koit replied:
Following your suggestion, I started to create the description of such a package, but then realised there is no good place for that because it violates two different rules at the same time. That's because my current approach is to have rule 1 for root METS file and rule 2 for representation METS. There might be value in having a package for the two combined, but that would go against our principle of violating only one rule at a time and open the door to a large number of other cases where violating several rules might seem reasonable. I think that would divert our attention away from vital stuff to "nice-to-have," so I'd avoid it 🙂.

PhillipTommerholt replied:
I would say it still just violates 1 rule and does it twice - but okay ;-)

PhillipTommerholt reviewed (changes requested):
Looks good and if we create a package with errors at both root and representation level then this can be used to test and give feedback to this issue: https://github.com/E-ARK-Software/py-rest-ip-validator/issues/17

koit committed:
Update testCase.xml

Shorten package names.
Add paths.
Fix spec version.

koit committed:
Move rules 4 and 5 to CSIP5 testCase.xml

Moved the rules because CSIP5 is the proper place to check for the existence of mets/@csip:OTHERCONTENTINFORMATIONTYPE.

PhillipTommerholt reviewed (changes requested):
version number in line 2 is 2.0.3 and not 2.0.4?

carlwilson committed:
Merge branch 'integration' into csip/mets-root/csip4
